src/enclave/rpc_sessions.h

```diff
@@ -420,14 +442,40 @@ namespace ccf
             return;
           }
 
-          search->second.second->recv(body.data, body.size);
+          search->second.second->recv(body.data, body.size, {});
         });
 
       DISPATCHER_SET_MESSAGE_HANDLER(
         disp, tls::tls_close, [this](const uint8_t* data, size_t size) {
           auto [id] = ringbuffer::read_message<tls::tls_close>(data, size);
           remove_session(id);
         });
+
+      DISPATCHER_SET_MESSAGE_HANDLER(
+        disp, quic::quic_start, [this](const uint8_t* data, size_t size) {
+          auto [new_id, listen_interface_name] =
+            ringbuffer::read_message<quic::quic_start>(data, size);
+          accept(new_id, listen_interface_name, true);
+          // UDP sessions are never removed because there is no connection
```

Comment:
Does this imply infinite memory growth inside the enclave then?

Reply:
Yes, because `remove_session` isn't called. This is the weakest point of the current experiment, and one of the things I want to discuss in a meeting.

One way to get around this is to call accept only once (ex. with a static flag that, once set, ignores new connections). This is a hack, but works quickly and easily. This is not a long term solution, but works because we only have an echo server for now.

Because UDP has no connection, we can continue that pattern (with a better implementation) for the foreseeable future. UDP packets don't start/stop connections, they always have src/dst addresses and the connection is always open, so a single HTTP 3 Endpoint is what we want, really.

Another way is to detect QUIC "streams". Those are handled by the enclave side and know when a stream has finished its work, but it has to tell the host to remove the connection. So, calling `accept` would be a reaction to a host event (new address not in the open list), while calling `remove_session` would be a reaction to an enclave event (QUIC telling us the stream is closing).

Because these don't come from the same side, we'll need some plumbing. Furthermore, there are many ways in which a QUIC stream may die, including not sending another packet. If it turns out is common for clients to not bother ACKing packets, servers need to be smarter to close the streams, and that may require even more plumbing.

So, the original design is to have a single channel between host(udp) and enclave(quic), and let the QUIC layer handle sessions, streams and memory lifetime.

Reply:
I wonder if it's worth creating "sessions" for UDP at all at this point? We don't want the same session limit behaviour, especially if we're not cleaning up (IIUC, we're creating a _new_ session for every incoming packet, so the session counts are really packet counts?).

In that model, we'd remove `quic_start` entirely, and in `quic_inbound` we'd dispatch to a single QuicHandler/QuicEndpoint instance. Internally, once that's done some QUIC parsing, it might be able to tell us about meaningful sessions/streams/connections that we want to count, or limit. But I don't think it's helpful to pair those to TLS connections before we know that?

(Note: I think it's fine to leave this as-is for this PR, and explore this separately in future)

Reply:
At the very least, make sure we only create the endpoint once

Reply:
Couple of points here:
* We need to create one UDP session per listening interface. If there are more than one UDP interface listening (different ports, IPs, etc), we need one `ConnID` for each and the ring-buffer needs to differentiate between them.
* The whole communication between host and enclave is based on `ConnID` and interface names and refactoring that now would be a prohibitive cost for the time being.
* In the end, the memory doesn't grow, because the `start` behaviour is only called once per UDP server listen, not once per client connection, so the right number of sessions/endpoints are created.